nxc/modules/defender.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+from sys import exit
+from impacket.dcerpc.v5 import rrp
+from impacket.examples.secretsdump import RemoteOperations
+
+# Registry keys for enabling Windows Defender
+ENABLE_REGISTRY_KEYS = {
+    "SOFTWARE\\Policies\\Microsoft\\Windows Defender": {
+        "PUAProtection": 1,
+        "ServiceKeepAlive": 1,
+        "DisableRoutinelyTakingAction": 0,
+        "DisableAntiSpyware": 0,
+        "DisableAntiVirus": 0
+    },
+    "SOFTWARE\\Policies\\Microsoft\\Windows Defender\\Real-Time Protection": {
+        "RealtimeScanDirection": 0,
+        "IOAVMaxSize": 0,
+        "DisableScanOnRealtimeEnable": 0,
+        "DisableRealtimeMonitoring": 0,
+        "DisableOnAccessProtection": 0,
+        "DisableIOAVProtection": 0,
+        "DisableBehaviorMonitoring": 0,
+        "DisableScriptScanning": 0,
+        "DisableIntrusionPreventionSystem": 0
+    }
+}
+
+# Registry keys for disabling Windows Defender
+DISABLE_REGISTRY_KEYS = {
+    "SOFTWARE\\Policies\\Microsoft\\Windows Defender": {
+        "PUAProtection": 0,
+        "ServiceKeepAlive": 0,
+        "DisableRoutinelyTakingAction": 1,
+        "DisableAntiSpyware": 1,
+        "DisableAntiVirus": 1
+    },
+    "SOFTWARE\\Policies\\Microsoft\\Windows Defender\\Real-Time Protection": {
+        "RealtimeScanDirection": 1,
+        "IOAVMaxSize": 1,
+        "DisableScanOnRealtimeEnable": 1,
+        "DisableRealtimeMonitoring": 1,
+        "DisableOnAccessProtection": 1,
+        "DisableIOAVProtection": 1,
+        "DisableBehaviorMonitoring": 1,
+        "DisableScriptScanning": 1,
+        "DisableIntrusionPreventionSystem": 1
+    }
+}
+# Reference
+# https://admx.help/HKLM/Software/Policies/Microsoft/Windows%20Defender
+
+class NXCModule:
+    """Defender by @byinarie"""
+    name = "defender"
+    description = "Enables/Disables Windows Defender"
+    supported_protocols = ["smb"]
+    opsec_safe = False
+    multiple_hosts = True
```

Comment:
Do this module should really be used on multiple hosts ?

Reply:
If you're trying to mass-pwn a network and want to disable Defender first I'd say so. Honestly we should probably remove the whole "multiple_hosts" warning, it doesn't make much sense to me.